src/main/resources/org/cbioportal/legacy/persistence/mybatis/StudyMapper.xml

```diff
@@ -178,4 +178,98 @@
+        SELECT
+            rs.RESOURCE_ID AS resourceId,
+            DISPLAY_NAME AS displayName,
+            rd.DESCRIPTION AS description,
+            rd.RESOURCE_TYPE AS resourceType,
+            rd.PRIORITY AS priority,
+            rd.OPEN_BY_DEFAULT AS openByDefault,
+            CANCER_STUDY_IDENTIFIER as cancerStudyIdentifier,
+            rd.CUSTOM_METADATA AS customMetaData,
+            COUNT(DISTINCT rs.INTERNAL_ID) AS sampleCount,
+            COUNT(DISTINCT p.INTERNAL_ID) AS patientCount
+        FROM resource_sample AS rs
+                 INNER JOIN resource_definition rd ON rd.RESOURCE_ID = rs.RESOURCE_ID
+                 INNER JOIN sample s ON s.INTERNAL_ID = rs.INTERNAL_ID
+                 INNER JOIN patient p ON p.INTERNAL_ID = s.PATIENT_ID
+                 INNER JOIN cancer_study cs ON cs.CANCER_STUDY_ID = p.CANCER_STUDY_ID
+        GROUP BY CANCER_STUDY_IDENTIFIER, DISPLAY_NAME, rs.RESOURCE_ID
+        UNION ALL
+        SELECT
+            rp.RESOURCE_ID AS resourceId,
+            DISPLAY_NAME AS displayName,
+            rd.DESCRIPTION AS description,
+            rd.RESOURCE_TYPE AS resourceType,
+            rd.PRIORITY AS priority,
+            rd.OPEN_BY_DEFAULT AS openByDefault,
+            CANCER_STUDY_IDENTIFIER as cancerStudyIdentifier,
+            rd.CUSTOM_METADATA AS customMetaData,
+            COUNT(DISTINCT s.INTERNAL_ID) AS sampleCount,
+            COUNT(DISTINCT rp.INTERNAL_ID) AS patientCount
+        FROM cancer_study cs
+                 INNER JOIN patient p ON cs.CANCER_STUDY_ID = p.CANCER_STUDY_ID
+                 LEFT JOIN sample s ON p.INTERNAL_ID = s.PATIENT_ID
+                 INNER JOIN resource_patient rp ON rp.INTERNAL_ID = s.PATIENT_ID
+                 INNER JOIN resource_definition rd ON rd.RESOURCE_ID = rp.RESOURCE_ID
+        GROUP BY CANCER_STUDY_IDENTIFIER, DISPLAY_NAME, rp.RESOURCE_ID
+        ORDER BY resourceId
+    </select>
+
+    <select id="getResourceCounts" resultType="org.cbioportal.legacy.model.ResourceCount">
+        SELECT
+            rs.RESOURCE_ID AS resourceId,
+            DISPLAY_NAME AS displayName,
+            rd.DESCRIPTION AS description,
+            rd.RESOURCE_TYPE AS resourceType,
+            rd.PRIORITY AS priority,
+            rd.OPEN_BY_DEFAULT AS openByDefault,
+            CANCER_STUDY_IDENTIFIER as cancerStudyIdentifier,
+            rd.CUSTOM_METADATA AS customMetaData,
+            COUNT(DISTINCT rs.INTERNAL_ID) AS sampleCount,
+            COUNT(DISTINCT p.INTERNAL_ID) AS patientCount
+        FROM resource_sample AS rs
+                 INNER JOIN resource_definition rd ON rd.RESOURCE_ID = rs.RESOURCE_ID
+                 INNER JOIN sample s ON s.INTERNAL_ID = rs.INTERNAL_ID
+                 INNER JOIN patient p ON p.INTERNAL_ID = s.PATIENT_ID
+                 INNER JOIN cancer_study cs ON cs.CANCER_STUDY_ID = p.CANCER_STUDY_ID
+        <where>
+            <if test="studyIds != null and studyIds.size > 0">
```

Comment:
```suggestion
            <if test="studyIds != null and !studyIds.isEmpty()">
```
We can just use isEmpty() function to see if it has any elements here, this can also apply to line#264